DOC: more complete gantt example (#6495)
mermaid-js/mermaid: packages/mermaid/src/docs/intro/syntax-reference.md

## Syntax Structure

One would notice that all **Diagrams definitions begin** with a declaration of the **diagram type**, followed by the definitions of the diagram and its contents. This declaration notifies the parser which kind of diagram the code is supposed to generate. The only exception to this a [Frontmatter](#frontmatter-for-diagram-code) configuration.

> Code is case insensitive. Indents are completely ignored but recommended for structure and readability.

Reviewer: ```suggestion
```

This is not a general statement we can make. There are case & indentation sensitive diagrams.

Line comments can ignore anything on the line after '%% '.

> Unknown words and misspellings will break a diagram, while parameters silently fail. Strings shouldn't be quoted and will parse based on expected parameters.

Reviewer: Strings have different quoting requirements in different diagrams. 

```suggestion
Unknown words and misspellings will break a diagram, while parameters silently fail.
```

Here you can edit certain values to change the behavior and appearance of the diagram.

Each of these techniques are functionally equivalent, but better for different deployments.

### [The initialize() call](./getting-started.md#_3-calling-the-javascript-api)

Used when Mermaid is called via an API, or through a `<script>` tag.

### Frontmatter for diagram code

Frontmatter is the term for adding YAML metadata at the start of code. This allows for reconfiguration of a diagram before it is rendered. You can pass metadata Frontmatter with your definition by adding `---` to the lines before and after the definition. This 'triple dash' MUST be the only character on the first line.

> Frontmatter uses YAML coding syntax. It requires any indenting to be consistent, settings are case sensitive. Mermaid will silently ignore misspelling, but badly formed parameters will break the diagram. Strings will inconsistently require quotation marks.

Reviewer: String quoting is purely YAML syntax dependent.

```suggestion
Frontmatter uses YAML syntax. It requires any indentation to be consistent and settings are case sensitive. Mermaid will silently ignore misspelling, but badly formed parameters will break the diagram. 
```

```mermaid-example
---
title: Frontmatter Example
displayMode: compact
config:
  theme: forest
gantt:
    useWidth: 400
    compact: true
---
gantt
    section Waffle
        Iron  : 1982, 3y
        House : 1986, 3y
```

### [Directives](../config/directives.md)

> Note: After 2025, directives will stop working. See [pull request #4691](https://github.com/mermaid-js/mermaid/pull/6491)
> 

Reviewer: The directives will continue to work, they're just not the recommended method anymore. 

```suggestion
```
